js/recipe.js

```diff
@@ -1,27 +1,83 @@
 'use strict';
 
-function choose_url (data, proxyPattern) {
-    //first try to prepare local_url from recipe
-    var local_url = data["md-url_recipe"] !== undefined ? prepare_url(data["md-url_recipe"][0], data) : null;
-
-    var use_url_proxy = data["md-use_url_proxy"] !== undefined ? data["md-use_url_proxy"] : "0";
-
-    //use the proxyPattern
-    if (proxyPattern && use_url_proxy == "1") {
-        if (local_url) {
-            data["md-local-url"] = [];
-            data["md-local-url"].push(local_url);
-        }
-        var ref_local_url = prepare_url(proxyPattern, data);
-        if (ref_local_url) return ref_local_url;
+/**
+ * Function designed to call functions in given order.
+ * @param url
+ * @returns {*}
+ */
+function processExternalUrl(url) {
+  var newUrl;
+  var process_callbacks = Drupal.settings.preprocessExternalUrlCallbacks;
+
+  process_callbacks.forEach(function(item, i, process_callbacks) {
+    if (item === 'ting_proxy') {
```

Comment:
Better to call function as string.
E.g. 
newUrl = url;
if(typeof item === 'function') {
    newUrl = item(newUrl);
}
return newUrlș